rust/tests/optimize_test.rs

```diff
@@ -0,0 +1,344 @@
+mod optimize {
+    extern crate deltalake;
+
+    use arrow::datatypes::Schema as ArrowSchema;
+    use arrow::{
+        array::{Int32Array, StringArray},
+        datatypes::{DataType, Field},
+        record_batch::RecordBatch,
+    };
+    use deltalake::optimize::{MetricDetails, Metrics};
+    use deltalake::writer::DeltaWriterError;
+    use deltalake::{
+        action, get_backend_for_uri_with_options,
+        optimize::Optimize,
+        writer::{DeltaWriter, RecordBatchWriter},
+        DeltaTableConfig, DeltaTableMetaData, PartitionFilter,
+    };
+    use deltalake::{DeltaTable, Schema, SchemaDataType, SchemaField};
+    use rand::prelude::*;
+    use serde_json::{Map, Value};
+    use std::{collections::HashMap, error::Error, sync::Arc};
+    use tempdir::TempDir;
+    struct Context {
+        pub tmp_dir: TempDir,
+        pub table: DeltaTable,
+    }
+
+    async fn setup_test(partitioned: bool) -> Result<Context, Box<dyn Error>> {
+        let schema = Schema::new(vec![
```

Comment:
Maybe worth saving this schema as a constant and then calling `try_into()` on it below to get the equivalent Arrow schema?

Reply:
I think this is okay for now. We can always pull it out if we want to reuse it in the future.